### PR TITLE
Updated provisioning script doc to include lima overrides for macOS and Linux

### PR DESCRIPTION
**Pull request by gunamata**
Signed-off-by: Gunasekhar Matamalam <gunasekhar.matamalam@suse.com>

### CODE REVIEW

File: docs/how-to-guides/provisioning-scripts.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  <TabItem value="Linux">
+
+```
+~/.local/share/rancher-desktop/lima/_config/override.yaml
+```
+
+  </TabItem>
+</Tabs>
+
+- Write a provisioning script in the `override.yaml` file created in the previous step. For example, you can use the below script to create additional mounts.
```
Comment (jandubois) [inline docs/how-to-guides/provisioning-scripts.md:32]: This is not really a provisioning script; it is just overriding/modifying a lima configuration setting.

Provisioning scripts are actual script (under the `provision` tag), that are not interpreted by Lima, but executed by a shell inside the VM.

File: docs/how-to-guides/provisioning-scripts.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+EOF
+```
+
+- Note that files with a file extension of `.stop`, such as `wipe-data.stop`, can be executed _after Rancher Desktop shuts down its Kubernetes backend_ (if enabled). Such files will run within the same Rancher Desktop WSL context.
+
+Example flow for `.stop` files:
+- Stop `k3s`, `dockerd` or `containerd`
+- Run deprovisioning scripts
+
+**Important to note:** Scripts need to be saved with Unix line endings; line-ending conversion is not done; and files with DOS line endings may produce unexpected results.
```
Comment (jandubois) [inline docs/how-to-guides/provisioning-scripts.md:85]: I was wondering if you should mention that the Windows provisioning scripts cannot change the hard ulimits on WSL2 (see https://github.com/rancher-sandbox/docs.rancherdesktop.io/pull/51#issuecomment-1112517037).

File: docs/how-to-guides/provisioning-scripts.md (hunk 1 above)
Comment (gunamata) [reply docs/how-to-guides/provisioning-scripts.md]: Modified the wording around examples provided to distinguish between a provisioning script and a simple lima configuration setting change.

File: docs/how-to-guides/provisioning-scripts.md (hunk 2 above)
Comment (gunamata) [reply docs/how-to-guides/provisioning-scripts.md]: Added a note that provisioning scripts can't be used to change/control certain things.

**Review by jandubois — APPROVED** [review]

**Pull request merged**